diff --git a/CMakeLists.txt b/CMakeLists.txt
index c8843bd0..e8311c48 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -31,6 +31,9 @@ INCLUDE(cmake/add_to_set.cmake)
 #
 # find_CHOLMOD
 INCLUDE(cmake/find_cholmod.cmake)
+#
+# CHECK_CXX_SOURCE_COMPILES
+INCLUDE(CheckCXXSourceCompiles)
 # ----------------------------------------------------------------------------
 # required_definition( variable type docstring )
 #
@@ -97,7 +100,7 @@ MACRO( optional_definition variable type docstring )
    ENDIF( ${type} STREQUAL "PATH" )
 ENDMACRO( optional_definition )
 # ============================================================================
-CMAKE_MINIMUM_REQUIRED(VERSION 3.5)
+CMAKE_MINIMUM_REQUIRED(VERSION 3.10)
 PROJECT(cppad_mixed)
 SET(cppad_mixed_version "2026.0.0")
 INCLUDE_DIRECTORIES( ${cppad_mixed_SOURCE_DIR}/include )
@@ -194,7 +197,7 @@ IF( extra_cxx_flags )
    int main(void)
    {  return 0; }"
    )
-   CHECK_CXX_SOURCE_RUNS("${source}" extra_cxx_flags_ok)
+   CHECK_CXX_SOURCE_COMPILES("${source}" extra_cxx_flags_ok)
    IF( NOT extra_cxx_flags_ok )
       MESSAGE(STATUS "Cannot compile a simple C++ program, extra_cxx_flags")
    ENDIF( NOT extra_cxx_flags_ok )
@@ -216,7 +219,7 @@ IF( extra_cxx_flags )
 ElSE( extra_cxx_flags )
    SET(CMAKE_REQUIRED_FLAGS "" )
 ENDIF( extra_cxx_flags )
-CHECK_CXX_SOURCE_RUNS("${source}" cppad_mixed_has_nullptr_flag)
+CHECK_CXX_SOURCE_COMPILES("${source}" cppad_mixed_has_nullptr_flag)
 IF( cppad_mixed_has_nullptr_flag )
    SET(cppad_mixed_has_nullptr_01 1)
 ELSE( cppad_mixed_has_nullptr_flag )
diff --git a/example/private/sparse_low2sym.cpp b/example/private/sparse_low2sym.cpp
index 51d4c974..a2c6f0fc 100644
--- a/example/private/sparse_low2sym.cpp
+++ b/example/private/sparse_low2sym.cpp
@@ -21,6 +21,7 @@ This example is not part of the
 {xrst_end sparse_low2sym.cpp}
 */
 // BEGIN C++
+# include <cassert>
 # include <cmath>
 # include <cstddef>
 # include <limits>
diff --git a/include/cppad/mixed/sparse_mat2low.hpp b/include/cppad/mixed/sparse_mat2low.hpp
index adf0f2c1..12387b20 100644
--- a/include/cppad/mixed/sparse_mat2low.hpp
+++ b/include/cppad/mixed/sparse_mat2low.hpp
@@ -64,6 +64,7 @@ and test of ``sparse_mat2low`` .
 
 {xrst_end sparse_mat2low}
 */
+# include <cassert>
 # include <Eigen/SparseCore>
 
 namespace CppAD { namespace mixed {
diff --git a/src/eigen/sparse_low_tri_sol.cpp b/src/eigen/sparse_low_tri_sol.cpp
index 35de187f..4320a925 100644
--- a/src/eigen/sparse_low_tri_sol.cpp
+++ b/src/eigen/sparse_low_tri_sol.cpp
@@ -56,6 +56,7 @@ and test of ``sparse_low_tri_sol`` .
 
 {xrst_end sparse_low_tri_sol}
 */
+# include <cassert>
 # include <cppad/mixed/sparse_low_tri_sol.hpp>
 # include <iostream>
 
diff --git a/src/eigen/sparse_up_tri_sol.cpp b/src/eigen/sparse_up_tri_sol.cpp
index a456da6b..7f2778a9 100644
--- a/src/eigen/sparse_up_tri_sol.cpp
+++ b/src/eigen/sparse_up_tri_sol.cpp
@@ -56,6 +56,7 @@ and test of ``sparse_up_tri_sol`` .
 
 {xrst_end sparse_up_tri_sol}
 */
+# include <cassert>
 # include <cppad/mixed/sparse_up_tri_sol.hpp>
 # include <iostream>
 
diff --git a/src/eigen/undetermined.cpp b/src/eigen/undetermined.cpp
index 80cf477e..10e7e932 100644
--- a/src/eigen/undetermined.cpp
+++ b/src/eigen/undetermined.cpp
@@ -152,6 +152,7 @@ to convert this (and :ref:`sample_fixed-name` ) to all sparse matrices.
 */
 
 
+# include <cassert>
 # include <cmath>
 # include <iostream>
 # include <Eigen/Core>
